src/Psalm/Internal/Analyzer/Statements/Expression/Call/ArgumentAnalyzer.php

```diff
@@ -1628,4 +1635,34 @@ private static function processTaintedness(
 
         return $input_type;
     }
+
+    private static function extendKeyedArrayComparison(Union $param_type, Union $input_type): string
+    {
+        // Default value is an empty string
+        $extension = '';
+
+        // Not sure if there's a better or more robust way to do this
+        $param_types = $param_type->getAtomicTypes();
+        $input_types = $input_type->getAtomicTypes();
+
+        if (!isset($param_types['array'], $input_types['array'])) {
+            return $extension;
+        }
+
+        $first_param_type = $param_type->getAtomicTypes()['array'];
+        $first_input_type = $input_type->getAtomicTypes()['array'];
+        if (!$first_param_type instanceof TKeyedArray || !$first_input_type instanceof TKeyedArray) {
+            return $extension;
+        }
+
+        // There's many ways to illustrate this but this is the simplest and provides info
+        // without being too opinionated
+        $extension .= '. The differences are in the following keys: ';
+        $param_keys = array_keys($first_param_type->properties);
```

Comment:
Sometimes, the difference comes with the fact that some keys are always defined but some are possibly undefined.

This is represented with a `?` after the key, it could be interesting to allow this feature to display that. It adds complexity, but it may be worth it.

For example:
`array{a: string, b?: int}` vs `array{a: string, b: int}` would display those keys as different `b, b?`

Reply:
Oh, that's a good idea as well!
I think the message needs to be careful to avoid implying that the keys mentioned are the only ones that are potentially wrong. Without testing, I think this will generate a confusing message for `array{a: string}` contained by `array{a: int, b: string}`. I would say maybe the message should be more specific and say "these keys are missing", and just ignore sealed arrays for now (ie extra keys on the child type are fine).